compiler-plugin/src/main/java/io/ballerina/lib/ai/np/compilerplugin/PromptGenerator.java

```diff
@@ -0,0 +1,271 @@
+                - Whenever you have a Json variable, NEVER access or manipulate Json variables. ALWAYS define a
+                record and convert the Json to that record and use it.
+                - When invoking resource function from a client, use the correct paths with accessor and paramters.
+                (eg: exampleClient->/path1/["param"]/path2.get(key="value"))
+                - When you are accessing a field of a record, always assign it into new variable and use that
+                variable in the next statement.
+                - Avoid long comments in the code. Use // for single line comments.
+                - Always use named arguments when providing values to any parameter. (eg: .get(key="value"))
+                - Mention types EXPLICITLY in variable declarations and foreach statements.
+                - Do not modify the README.md file unless asked to be modified explicitly in the query.
+                - Do not add/modify toml files(Config.toml/Ballerina.toml) unless asked.
+                - In the library API documentation if the service type is specified as generic, adhere to the
+                instructions specified there on writing the service.
+                - For GraphQL service related queries, If the user haven't specified their own GraphQL Scehma,
+                Write the proposed GraphQL schema for the user query right after explanation before generating the
+                ballerina code. Use same names as the GraphQL Schema when defining record types.
+                                
+                Begin your response with the explanation, once the entire explanation is finished only, include
+                codeblock segments(if any) in the end of the response.
+                The explanation should explain the control flow decided in step 2, along with the selected
+                libraries and their functions.
+                                
+                Each file which needs modifications, should have a codeblock segment and it MUST have complete
+                file content with the proposed change.
+                The codeblock segments should only have .bal contents and it should not generate or modify any
+                other file types. Politely decline if the query requests for such cases.
+                                
+                Example Codeblock segment:
+                <code filename="main.bal">
+                \\`\\`\\`ballerina
+                //code goes here
+                \\`\\`\\`
+                </code>
+                """.formatted(LANG_LIBS);
+    }
+
+    public static String getUserPrompt(
+            String useCase,
+            JsonArray existingCode
+    ) {
+        return """
+                QUERY: The query you need to answer using the provided api documentation.
+                <query>
+                %s
+                </query>
+
+                Existing Code: Users existing code.
```

Comment:
```suggestion
                Existing Code: User's existing code.
```

Please fix up the prompts.

Reply:
This prompts are not completed yet.
, that is one of reason for mark this PR as a draft PR